dependencies.yaml

```diff
@@ -234,12 +234,10 @@ dependencies:
         packages:
           - ipython
           - nbsphinx
-          - numpydoc>=1.5
+          - &numpydoc numpydoc>=1.7
           - pydata-sphinx-theme
           - recommonmark
-          # TODO: sphinx upper version limit can likely be removed after pydata 0.13
-          # is released. See https://github.com/pydata/pydata-sphinx-theme/issues/1094
-          - sphinx<6
+          - sphinx>=8.0.0
```

Comment:
As noted in the comment, this was pinned to an old version of `sphinx` in PR ( https://github.com/rapidsai/cucim/pull/481 ) to workaround this upstream `pydata-sphinx-theme` issue ( https://github.com/pydata/pydata-sphinx-theme/issues/1094 )

Looks like that issue was subsequently fixed in PR ( https://github.com/pydata/pydata-sphinx-theme/pull/1097 ) as part of `pydata-sphinx-theme`'s `0.13.0` release

Given that has been out for a while (and several versions have since been released), think we are ok dropping it